src/fontra/views/fontinfo/panel-development-status-definitions.js

```diff
@@ -0,0 +1,325 @@
+  grid-row-gap: 0.1em;
+  grid-column-gap: 1em;
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-delete {
+  justify-self: end;
+  /*align-self: start;*/
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-color {
+  width: 8em;
+  /*height: 1.6em; without height: best compromise for all browsers*/
+  margin: 0;
+  padding: 0;
+  outline: none;
+  border: none;
+  border-color: transparent;
+  border-radius: 0.25em;
+}
+`);
+
+class StatusDefBox extends HTMLElement {
+  constructor(fontController, statusIndex, postChange, setupUI) {
+    super();
+    this.classList.add("fontra-ui-font-info-status-definitions-panel-status-def-box");
+    this.fontController = fontController;
+    this.statusIndex = statusIndex;
+    this.postChange = postChange;
+    this.setupUI = setupUI;
+    this._updateContents();
+  }
+
+  get statusDef() {
+    return this.fontController.customData["fontra.sourceStatusFieldDefinitions"][
+      this.statusIndex
+    ];
+  }
+
+  replaceStatusDef(newStatusDef, undoLabel, statusIndex = this.statusIndex) {
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"][statusIndex] =
+        newStatusDef;
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+
+  deleteStatusDef(statusIndex) {
+    const undoLabel = `delete status def '${this.statusDef.name}'`;
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"].splice(statusIndex, 1);
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+
+  changeStatusDefIsDefault(statusDef, event) {
+    const undoLabel = `change status definition isDefault`;
+    if (!event.target.checked) {
+      delete statusDef["isDefault"];
+      this.replaceStatusDef(statusDef, undoLabel);
+      return;
+    }
+
+    // If checked: Set all status definitions to false, first.
+    for (const [index, statusDefTemp] of enumerate(
+      this.fontController.customData["fontra.sourceStatusFieldDefinitions"]
+    )) {
+      delete statusDefTemp["isDefault"];
+      this.replaceStatusDef(statusDefTemp, undoLabel, index);
+    }
+    // Then: set this status definition to true
+    const updatedStatusDef = {
+      ...statusDef,
+      isDefault: true,
+    };
+    this.replaceStatusDef(updatedStatusDef, undoLabel);
+  }
+
+  _updateContents() {
+    this.innerHTML = "";
+    const statusDef = this.statusDef;
+    this.append(
+      html.input({
+        "type": "color",
+        "class": "fontra-ui-font-info-status-definitions-panel-status-def-box-color",
+        "value": rgbaToHex(statusDef.color),
```

Comment:
I read that the color input does not support opacity, and that the value string _must_ be a 7-char string: `#FF9988`. However, `rgbaToHex` can return a 9-char string if opcaticy isn't 1. If we want to not support opacity (which would be totally okay), we need to make sure to drop any opacity. Perhaps like this:
```js
   statusDef.color.slice(0, 3).concat([1])
```

Reply:
Good point. Thanks for the code snipped. Much appreciated.